engine/runtime-parser-dsl/src/main/java/org/enso/runtime/parser/dsl/GenerateIR.java

```diff
@@ -0,0 +1,31 @@
+package org.enso.runtime.parser.dsl;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A class annotated with this annotation will be processed by the IR processor. The processor will
+ * generate a super class that is meant to be extended by this class. The generated class will have
+ * the same package as this class, and its name will have the "Gen" suffix. Majority of the methods
```

Comment:
It is possible to extract the name of the super class from `class X extends MyParent` definition. E.g. then the name of the super class could be arbitrary. It would have to be in the same package, but there would be no restriction on the `Gen` suffix.

Reply:
Implemented and tested in bdd73c351ab1b778a92ce279e7be0596f8bda6f3